src/inference/include/openvino/runtime/core.hpp

```diff
@@ -84,6 +84,12 @@ class OPENVINO_RUNTIME_API Core {
     std::shared_ptr<ov::Model> read_model(const std::wstring& model_path,
                                           const std::wstring& bin_path = {},
                                           const ov::AnyMap& properties = {}) const;
+    template <class Path,
```

Comment:
This should not required and it could be modified in previous function using `if constexpr` instead another template function